### PR TITLE
Support mesh config ingressSelector

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/24055

API: https://github.com/istio/api/pull/1446 will be WIP until that is merged

### CODE REVIEW

File: pilot/pkg/config/kube/ingress/controller.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ func (c *controller) List(typ resource.GroupVersionKind, namespace string) ([]mo
 		case virtualServiceGvk:
 			ConvertIngressVirtualService(*ingress, c.domainSuffix, ingressByHost)
 		case gatewayGvk:
-			gateways := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
+			gateways := ConvertIngressV1alpha3(*ingress, c.mesh.IngressSelector, c.domainSuffix)
```
Comment (costinm) [inline pilot/pkg/config/kube/ingress/controller.go:322]: Nit: I would just pass c.mesh.

File: pilot/pkg/config/kube/ingress/conversion.go [hunk 2]
```diff
@@ -68,10 +68,18 @@ func decodeIngressRuleName(name string) (ingressName string, ruleNum, pathNum in
...
 // ConvertIngressV1alpha3 converts from ingress spec to Istio Gateway
-func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) model.Config {
-	gateway := &networking.Gateway{
-		Selector: labels.Instance{constants.IstioLabel: constants.IstioIngressLabelValue},
+func ConvertIngressV1alpha3(ingress v1beta1.Ingress, selector string, domainSuffix string) model.Config {
+	gateway := &networking.Gateway{}
+	if len(selector) > 0 {
+		gateway.Selector = labels.Instance{constants.IstioLabel: selector}
+	} else {
+		gateway.Selector = defaultSelector
```
Comment (costinm) [inline pilot/pkg/config/kube/ingress/conversion.go:82]: If you could pass ingressService - maybe you can default to that value, unless it is "ingressgateway".

File: pilot/pkg/config/kube/ingress/conversion.go [hunk 3]
```diff
@@ -68,10 +68,18 @@ func decodeIngressRuleName(name string) (ingressName string, ruleNum, pathNum in
...
+var defaultSelector = labels.Instance{constants.IstioLabel: constants.IstioIngressLabelValue}
+
 // ConvertIngressV1alpha3 converts from ingress spec to Istio Gateway
-func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) model.Config {
-	gateway := &networking.Gateway{
-		Selector: labels.Instance{constants.IstioLabel: constants.IstioIngressLabelValue},
+func ConvertIngressV1alpha3(ingress v1beta1.Ingress, selector string, domainSuffix string) model.Config {
+	gateway := &networking.Gateway{}
+	if len(selector) > 0 {
+		gateway.Selector = labels.Instance{constants.IstioLabel: selector}
```
Comment (costinm) [inline pilot/pkg/config/kube/ingress/conversion.go:80]: If you pass mesh or ingressService - please warn if they are different.

File: pilot/pkg/config/kube/ingress/conversion.go (hunk 2 above)
Comment (howardjohn) [reply pilot/pkg/config/kube/ingress/conversion.go:89]: done

**Pull request merged**